### PR TITLE
properly fix migration in mem ack aggregation

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Fix migration in mem ack aggregation, since sender based / receiver based aggregation cannot be unified

**Why?**
Correctly unify sender based & receiver based ack aggregation

**How did you test it?**
UT

**Potential risks**
N/A

**Is hotfix candidate?**
1.21.2

### CODE REVIEW

File: service/history/shard/context_impl.go [hunk 1]
```diff
@@ -383,16 +383,23 @@ func (s *ContextImpl) UpdateReplicationQueueReaderState(
...
-	remoteClusterInfo.AckedReplicationTimestamps[s.shardID] = ackTimestamp
+	clusterInfo := s.clusterMetadata.GetAllClusterInfo()
+	remoteClusterInfo := s.getOrUpdateRemoteClusterInfoLocked(clusterName)
+	for _, remoteShardID := range common.MapShardID(
+		clusterInfo[s.clusterMetadata.GetCurrentClusterName()].ShardCount,
+		clusterInfo[clusterName].ShardCount,
+		s.shardID,
+	) {
+		remoteClusterInfo.AckedReplicationTaskIDs[remoteShardID] = ackTaskID
+		remoteClusterInfo.AckedReplicationTimestamps[remoteShardID] = ackTimestamp
```
Comment (yux0) [inline service/history/shard/context_impl.go:401]: One thing I am not following, why the acked task id can be used across different share ids?

**Review by yux0 — APPROVED** [review]

File: service/history/shard/context_impl.go (hunk 1 above)
Comment (wxing1292) [reply service/history/shard/context_impl.go:401]: for polling mode:
targer cluster's 1 poller may poll & ack on behalf of >= 1 shards, here source cluster should update all relevant shard's ack task IDs

for streaming mode:
each target shard will has its own grpc stream & update ack using reader ID (cluster ID + shard ID)

**Comment by alexshtin** [comment]
Is it hotfix candidate or not? If not, remove 1.21.2 tag, if yes, please add it to description.

**Comment by wxing1292** [comment]
> Is it hotfix candidate or not? If not, remove 1.21.2 tag, if yes, please add it to description.

this PR is, take a look at the github tag

**Pull request merged**